automatic-curtain/cpp/automatic-curtain.cpp

```diff
@@ -250,31 +280,9 @@ int main() {
 
 #ifndef SIMULATE_DEVICES
 
-  string unknownPlatformMessage = "This sample uses the MRAA/UPM library for I/O access, "
-      "you are running it on an unrecognized platform. "
-      "You may need to modify the MRAA/UPM initialization code to "
-      "ensure it works properly on your platform.\n\n";
-  //Update these values as per the hardware board being used.
-  int aPinRotary = 1,
-      aPinLight = 2,
-      dPinButton = 4,
-      i2cPort = 0;
-  // check which board we are running on
-  Platform platform = getPlatformType();
-  switch (platform) {
-    case INTEL_UP2:
-#ifdef USING_GROVE_PI_SHIELD //512 offset needed for the shield
-      aPinRotary += 512;     // A1 Connector
-      aPinLight += 512;      // A2 Connector
-      dPinButton += 512;     // D4 Connector
-      break;
-#else
-      cerr << "Not using Grove provide your pinout here" << endl;
-      return -1;
-#endif
-      default:
-          cerr << unknownPlatformMessage;
-  }
+  int aPinRotary, aPinLight, dPinButton, i2cPort;
+  if (initPlatform(aPinRotary, aPinLight, dPinButton, i2cPort) == -1)
+	  return -1;
```

Comment:
While I prefer error codes in tools development, here it is not helpful because how would user know what does -1 mean? I think instead of that a verbose error message would be better since usability ease is a key criteria.

Reply:
Error message + code is likely best here, also consistent.